docs/source/usersguide/input.rst

```diff
@@ -558,19 +559,127 @@ The following code snippet illustrates the instantiation of the ``CPUSolver`` fo
 Fixed Source Calculations
 -------------------------
 
-aljlkjadf
+It is also possible to add a fixed source to any region in OpenMOC. During computation of the total source, a fixed source is added together with the calculated scattering source and fission source for each flat source region. By default, the fixed source is set to zero everywhere. The ``setFixedSourceByFSR(...)`` routine allows the user to set the fixed source in a given flat source region using its unique ID. For most calcuations, setting the fixed source for every flat source region of interest individually can be cumbersome. In addition, this would require retreiving the unique ID for every flat source region in which the user desires to set the fixed source. Therefore, the ``setFixedSourceByCell(...)`` routine allows the user to set the fixed source for every flat source region within a cell to a common value. An example is given below for setting the fixed source of a ``Cell`` with a point source to unity in the first energy group.
+
+.. code-block:: python
+
+  # Set the fixed source inside the source cell in group 1 to unity
+  solver.setFixedSourceByCell(source_cell, 1, 1.0)
+
+
+The equivalent code for setting the point source of all flat source regions within the source cell using ``setFixedSourceByFSR(...)`` is given below.
+
+.. code-block:: python
+
+  # Get the unique ID of the cell containing the point source
+  point_source_cell_id = source_cell.getId()
+
+  # Loop over all FSRs and test if the FSR is within the point source cell
+  for fsr_id in xrange(solver.getGeometry().getNumFSRs()):
```

Comment:
I wouldn't use `xrange` here - although there is a good performance reason to use `xrange` over `range` in Python 2, that is no longer the case in Python 3.